src/node_buffer.cc

```diff
@@ -541,6 +659,121 @@ void WriteDoubleBE(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+template <typename T, enum Endianness endianness,
+         T (*strtoT)(const char*, char**, int)>
+uint32_t WriteInt64Generic(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  CHECK(args[0]->IsObject());
+  ARGS_THIS(args[0].As<Object>())
+
+  T val;
+  if (args[1]->IsNumber()) {
+    val = args[1]->IntegerValue();
+  } else if (args[1]->IsString()) {
+    // Have to do this because strt(u)oll doesn't set errno to 0 on success
+    errno = 0;
+    node::Utf8Value str(env->isolate(), args[1]);
+    val = strtoT(*str, nullptr, 0);
+
+    if (errno != 0) {
+      if (errno == EINVAL) {
+        env->ThrowTypeError("value is invalid");
+      } else if (errno == ERANGE) {
```

Comment:
EINVAL could be a CHECK, it means the base parameter is invalid (i.e. an application bug.)

ERANGE means the value-as-string is too large or too small to fit in the designated type (i.e. run-time error.)

Apropos the base, I think it would be better to pick a specific base.  0 means auto-detect and will try to interpret the string as octal, decimal or hexadecimal.  Anytime computers try to auto-detect something, there is a margin for error.  :-)